db/blip_sync_context.go

```diff
@@ -314,40 +320,44 @@ func (bsc *BlipSyncContext) sendRevisionWithProperties(sender *blip.Sender, docI
 	}
 	bsc.dbStats.StatsDatabase().Add(base.StatKeyNumDocReadsBlip, 1)
 
-	base.Tracef(base.KeySync, "Sending revision %s/%s, body:%s, properties: %v, attDigests: %v", base.UD(docID), revID, base.UD(string(bodyBytes)), base.UD(properties), attDigests)
+	base.TracefCtx(bsc.loggingCtx, base.KeySync, "Sending revision %s/%s, body:%s, properties: %v, attDigests: %v", base.UD(docID), revID, base.UD(string(bodyBytes)), base.UD(properties), attDigests)
 
 	// TODO: When CBG-881 is implemented, update this stat and SendRevCountError based on sendRev response
 	bsc.replicationStats.SendRevCount.Add(1)
 
-	if len(attDigests) > 0 {
+	// asyncronously wait for a response if we have attachment digests to verify, or if we have a registered callback.
+	verifyAttachments := len(attDigests) > 0
+	awaitResponse := responseCallback != nil || verifyAttachments
+
+	if verifyAttachments {
```

Comment:
What do you think about moving this inside 'if awaitResponse', to reduce the chance of future regressions calling bsc.addAllowedAttachments and then not awaiting the response?